jenkins/opensearch-ruby/Jenkinsfile

```diff
@@ -0,0 +1,60 @@
+pipeline {
+    options {
+        timeout(time: 1, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+                name: 'OPENSEARCH_RUBY_RELEASE_BRANCH_NAME',
+                description: 'Branch name for the release',
+                trim: true
+        )
+    }
+    stages {
+        stage('opensearch-ruby-build') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
+                    image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211130'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    if (params.OPENSEARCH_RUBY_RELEASE_BRANCH_NAME == '') {
+                        currentBuild.result = 'ABORTED'
+                        error('OPENSEARCH_RUBY_RELEASE_BRANCH_NAME parameter is not set')
+                    }
+                    git url: 'https://github.com/opensearch-project/opensearch-ruby.git', branch: "$OPENSEARCH_RUBY_RELEASE_BRANCH_NAME"
+                    withCredentials([string(credentialsId: 'opensearchproject-rubygems-private-key', variable: 'RUBYGEMS_PRIVATE_KEY')]) {
+                        sh'''
+                            set +x
+                            echo "$RUBYGEMS_PRIVATE_KEY" > private_key.pem
```

Comment:
Rather than execute this as shell script, can we use Jenkins to setup the key in the proper location.  We need to be very careful about the accessibility of this key